Directory structure and some pertinent files

gregpoore opened the pull request.

ekopylova commented on cgc/Mapping_file_lucky_50.txt:
```diff
@@ -0,0 +1 @@
s47	AAAAAAAAAA	TTTTTTTTTT	UNCID_1781098.7dab0105-975d-459f-97d3-f1462afddd4d.sorted_genome_alignments.bam	Primary Tumor	RNA-Seq	FF1F5513-8463-4B1E-B254-72D9E0863D35
s48	AAAAAAAAAA	TTTTTTTTTT	TCGA-G2-A2EK-01A-22D-A18D_120913_SN208_0429_AC187FACXX_s_3_rg.sorted.bam	Primary Tumor	WGS	FF1691F2-DF11-4068-8D8D-A778F384EC60
s49	AAAAAAAAAA	TTTTTTTTTT	UNCID_1210996.bc993392-5f2f-4cb8-8d27-26f3df2b3583.sorted_genome_alignments.bam	Primary Tumor	RNA-Seq	FF1691F2-DF11-4068-8D8D-A778F384EC60
s50	AAAAAAAAAA	TTTTTTTTTT	8d9e0153be8bfe0c936ac4e30cbcbecd.bam	Blood Derived Normal	WGS	FF1500F7-8FC1-4ACD-B93D-375BF881B90A
```
I don't think we have info for BarcodeSequence, LinkerPrimerSequence, you can just leave them blank (as a tab).

ekopylova replied:
Just to verify, we'll have more columns in the metadata file for the full TCGA dataset right (e.g., all clinical metadata)?

gregpoore replied:
I see. The current ones were just fillers, so I can remove them. Also, there will be more metadata once the CGC API is figured out. All of these 50 had to be copied and pasted from the CGC interface, so I limited the amount of metadata. I'll update both of these things tomorrow!

ekopylova commented on metadata/make_merged_CDE_table.py:
```diff
@@ -0,0 +1,50 @@
+dfIntersect.to_csv('Intersected_CDEs.csv')
+dfFullJoin.to_csv('Full_Joined_CDEs.csv')
+
+print "Complete! Saved in Intersected_CDEs.csv"
```
missing a new line here

gregpoore replied:
Got it! I also should write that it creates the full-join file too.

wasade replied:
Is this Python 3?

On Aug 10, 2016 07:08, "Evguenia Kopylova" notifications@github.com wrote:

> In metadata/make_merged_CDE_table.py
> https://github.com/biocore/tcga/pull/1#discussion_r74182791:
> 
> > -   pd.read_csv(eachFile,
> > -       sep='\t',
> > -       usecols = intersection_headers_string)
> > -   for eachFile in allTransposedFiles)
> >   +
> >   +allTransposedFiles = glob.glob("./Transposed*.txt")
> >   +dfFullJoin = pd.concat(
> > -   pd.read_csv(eachFile,
> > -       sep='\t')
> > -   for eachFile in allTransposedFiles)
> >   +
> >   +## Write to csv file
> >   +dfIntersect.to_csv('Intersected_CDEs.csv')
> >   +dfFullJoin.to_csv('Full_Joined_CDEs.csv')
> >   +
> >   +print "Complete! Saved in Intersected_CDEs.csv"
> 
> missing a new line here
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/biocore/tcga/pull/1/files/0de2130298adbac726c4e3f9459c61fb12254054#r74182791,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAc8sm4d8cUAHiqi5S0dqE2e1AqnAsKTks5qeU7UgaJpZM4JgsdH
> .

ekopylova replied:
I was going to ask that too, generally we try to use Python 3 in new projects, however we'll be also using QIIME 1.9.1 for diversity analyses so wasn't sure yet how this would effect development in Python 3.

gregpoore replied:
@wasade @ekopylova This was written in Python 2.7. I'm relatively new to Python (previously R/Matlab/Perl/SQL), so I am not familiar with the specific syntax differences in porting one to the other. Does it have to be ported or can it be left as-is?

wasade replied:
In this specific line, the py3 change is that print is a function. In
general, the differences are minor with the exception of string handling.
There should be easy to find guides of what the differences are though
online. First good place to start is just to run the script with py3 :)

On Aug 11, 2016 01:19, "Greg Poore" notifications@github.com wrote:

> In metadata/make_merged_CDE_table.py
> https://github.com/biocore/tcga/pull/1#discussion_r74341947:
> 
> > -   pd.read_csv(eachFile,
> > -       sep='\t',
> > -       usecols = intersection_headers_string)
> > -   for eachFile in allTransposedFiles)
> >   +
> >   +allTransposedFiles = glob.glob("./Transposed*.txt")
> >   +dfFullJoin = pd.concat(
> > -   pd.read_csv(eachFile,
> > -       sep='\t')
> > -   for eachFile in allTransposedFiles)
> >   +
> >   +## Write to csv file
> >   +dfIntersect.to_csv('Intersected_CDEs.csv')
> >   +dfFullJoin.to_csv('Full_Joined_CDEs.csv')
> >   +
> >   +print "Complete! Saved in Intersected_CDEs.csv"
> 
> @wasade https://github.com/wasade @ekopylova
> https://github.com/ekopylova This was written in Python 2.7. I'm
> relatively new to Python (previously R/Matlab/Perl/SQL), so I am not
> familiar with the specific syntax differences in porting one to the other.
> Does it have to be ported or can it be left as-is?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/biocore/tcga/pull/1/files/0de2130298adbac726c4e3f9459c61fb12254054#r74341947,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAc8st7eLrU5R6U0nK7b7Pjyp2fXqLYIks5qek6BgaJpZM4JgsdH
> .

ekopylova commented on metadata/make_merged_CDE_table.py:
```diff
@@ -0,0 +1,50 @@
+
+## Call Bash script to transpose data
+print "Transposing CDE file..."
+subprocess.call("./text_transpose.sh", shell=True)
```
Best to avoid hardcoded paths. Could the path to the metadata directory be an argument to `make_merged_CDE_table.py`?

gregpoore replied:
I've included this recommendation in the list of changes for the new script!

ekopylova commented on python_scripts/make_merged_CDE_table.py:
```diff
@@ -0,0 +1,50 @@
+#!/bin/env python
```
Is it possible to write a unit test for this file? You can create a new `test` directory within `python_scripts` and put `test_make_merged_CDE_table.py` there. [Here](https://github.com/biocore/WGS-HGT/tree/master/benchmark/tests) are some example testing scripts and I'll also submit shortly a PR for testing the Kraken results to BIOM table script.

ekopylova replied:
@gregpoore are you planning to run QIIME metadata file generation on CGC or locally? I think it should be OK for us to use Python 3.

gregpoore replied:
@ekopylova I'll be running the QIIME metadata file generation locally. As long as the Python/R CGC API works, building the tables and associated mapping file will be relatively straightforward.

gregpoore commented:
Hi @ekopylova,

I reworked a bit of the Python script to:
1. Integrate the Bash commands directly into the Python script (no need for a separate .sh file)
2. Removed all hard-links
3. Added a getopts module to allow I/O arguments to be passed via the command line (note: must consist of full paths to the directories)
4. Added a progress bar for Pandas concatenation operations
5. Gave clearer onscreen explanations as the script operates and added a "Tips" comment at the top of the script describing its dependencies.

I'll work on making the test file after getting the CGC API to work.

Greg

ekopylova commented:
@gregpoore  sounds great, thanks!

Could you also update the code to use Python 3? You can create a conda environment for this project requirement Python 3 and any other software/library dependencies.

gregpoore commented:
@ekopylova I'll work on that after the API for the QIIME mapping file. Which brings me to another point: I tried doing it in CGC's R API today (more familiar for me than Python), but apparently there is a 100 file limit placed on the calls... I'm going to try again with Python tomorrow, but do note that their tutorial specifically states that not all of the code is compatible with Python 3.x (see "Requirements" section under http://docs.cancergenomicscloud.org/docs/api-quickstart). Jad and I also tried making a tool on CGC to extract out metadata with no other operations, but it took way too long (6-7 min for 4 BAM files). I'll try to get the Python/R API working tomorrow and we'll go from there.

ekopylova commented on python_scripts/cgc_python_api_by_disease_type.py:
```diff
@@ -0,0 +1,92 @@
+	# Grab list of files, their names and IDs
+	file_list = api.files.query(
+	    project=cgcUserName+'/'+cgcProjectName,
+	    #limit = 1000, # Based on API calling limit of 1000 per hour
```
has this issue been resolved? Are you still limited to 1000 files per hour?

gregpoore replied:
I haven't heard back from Gaurav yet, so yes, still limited. Technically, we each could pick a subset of the sub-1000 file cancers and run them on our own. There is a disease_type column, so the tables could just be concatenated later on. I'll ping Erik this time and copy another member at SB named Brandi to see if we can get an update out of them.

ekopylova commented:
@gregpoore  thanks for the new metadata parsing scripts. Could you write unit tests for these and we will merge this PR? PRs should focus on one issue at a time for rapid co-development (e.g., fast reviews, dependent code). If there are additional methods, you can create issues for them and resolve in the next PR.

gregpoore commented:
@ekopylova, I tested the scripts with Python 3.5 and they function normally. I'm working on writing the testing scripts now. Given that this is my first shot at this, your feedback will be helpful...

Sorry for the multiple file additions on one PR. This is my first time using GitHub extensively for code dev, so it's a learning experience!

gregpoore commented:
@ekopylova: I'm having some trouble with the test files but I'll keep working on it later on

wasade commented on docker/cgc_metadata_to_qiime_mapping_file.py:
```diff
@@ -0,0 +1,138 @@
+from tqdm import tqdm # progressbar
+from tqdm import trange
+
+parser = argparse.ArgumentParser(description="""
```
Is there a reason to use `argparse` over `click`? I ask because `biocore` in general has standardized on `click` for CLI interface

gregpoore replied:
`argparse` was mostly chosen out of convenience but also because various sources said it was the recommended CLI parsing module in the Python standard library. If it needs to be ported to `click`, it can be done.

ekopylova replied:
It should be relatively quick to update to click from your code, include `import click` together with your other libraries and here's an [example from my PR](https://github.com/biocore/tcga/pull/2/files#diff-9186f4fdc9d906eaca7baa402d356496R146) on how to use it. There's also the [click documentation](http://click.pocoo.org/5/) for other options.

wasade commented on docker/cgc_metadata_to_qiime_mapping_file.py:
```diff
@@ -0,0 +1,138 @@
+from tqdm import tqdm # progressbar
+from tqdm import trange
+
+parser = argparse.ArgumentParser(description="""
```
Doesn't need to be click but it is very clean and very simple especially
for nested command structures. Argparse is also, unfortunately, not POSIX
compliant

On Aug 24, 2016 12:33, "Greg Poore" notifications@github.com wrote:

> In docker/cgc_metadata_to_qiime_mapping_file.py
> https://github.com/biocore/tcga/pull/1#discussion_r76123428:
> 
> > +#!/usr/bin/env python3
> > +# ----------------------------------------------------------------------------
> > +# Copyright (c) 2016--, Gregory Poore.
> > +#
> > +# Distributed under the terms of the Modified BSD License.
> > +#
> > +# ----------------------------------------------------------------------------
> > +import numpy as np
> > +import pandas as pd
> > +import sevenbridges as sbg
> > +import glob, sys, getopt, os, json, unicodedata, argparse, time
> > +from pandas.io.json import json_normalize
> > +from tqdm import tqdm # progressbar
> > +from tqdm import trange
> > +
> > +parser = argparse.ArgumentParser(description="""
> 
> argparse was mostly chosen out of convenience but also because various
> sources said it was the recommended CLI parsing module in the Python
> standard library. If it needs to be ported to click, it can be done.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/biocore/tcga/pull/1/files/eab95f73fb3f0519e275159d1c0e1481762b94ef..dda326f55f4f59a7698f267f00698e5934ad1830#r76123428,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAc8siN348YRun_0uEg2CLeXkeNRcKaVks5qjJyKgaJpZM4JgsdH
> .
